h5py/api_functions.txt

```diff
@@ -241,12 +237,9 @@ hdf5:
   hid_t     H5Oopen_by_addr(hid_t loc_id, haddr_t addr)
   hid_t     H5Oopen_by_idx(hid_t loc_id, char *group_name, H5_index_t idx_type, H5_iter_order_t order, hsize_t n, hid_t lapl_id)
 
-  1.6.0-1.11.2 herr_t    H5Oget_info(hid_t loc_id, H5O_info_t *oinfo)
-  1.11.6 herr_t    H5Oget_info1(hid_t loc_id, H5O_info_t *oinfo)
-  1.6.0-1.11.2 herr_t    H5Oget_info_by_name(hid_t loc_id, char *name, H5O_info_t *oinfo, hid_t lapl_id)
-  1.11.6 herr_t    H5Oget_info_by_name1(hid_t loc_id, char *name, H5O_info_t *oinfo, hid_t lapl_id)
-  1.6.0-1.11.2 herr_t    H5Oget_info_by_idx(hid_t loc_id, char *group_name,  H5_index_t idx_type, H5_iter_order_t order, hsize_t n, H5O_info_t *oinfo, hid_t lapl_id)
-  1.11.6 herr_t    H5Oget_info_by_idx1(hid_t loc_id, char *group_name,  H5_index_t idx_type, H5_iter_order_t order, hsize_t n, H5O_info_t *oinfo, hid_t lapl_id)
+  herr_t    H5Oget_info(hid_t loc_id, H5O_info_t *oinfo)
```

Comment:
Are we still being saved by the caching on azure here?  This looks like an API that was changed for 1.12 so will escape the 1.8 related pinning.

Reply:
I don't think there's a problem being hidden by caching here. On the second commit of this PR, HDF5 1.12 was [built from source](https://dev.azure.com/h5pyappveyor/h5py/_build/results?buildId=462&view=logs&j=f3407ca5-1e8e-5831-d467-f40b15dcbaf4&t=96643522-c40b-5680-4fd9-7ab105c6ebd9). [Cache isolation](https://docs.microsoft.com/en-us/azure/devops/pipelines/release/caching?view=azure-devops#cache-isolation-and-security) seems to give PRs read-only access to the cache for master, so the cached build that caused problems in PR #1506 shouldn't affect any other PR.

[H5Oget_info](https://confluence.hdfgroup.org/display/HDF5/H5O_GET_INFO) was introduced in 1.8, and versioned in 1.12. So the duplication I'm removing was a necessary workaround when we were requesting the 1.6 API, because in 1.12 it would float to the newest version. It was added in #1335. Now that we're requesting the 1.8 API, 1.12 knows to give us the older version, so this simplification is possible.

There's one function, [H5Pencode](https://confluence.hdfgroup.org/display/HDF5/H5P_ENCODE), which was added in 1.10 and versioned in 1.12. We don't currently wrap that, but if we do, we'll need a similar workaround.

Reply:
Ah, I was inferring from the version spec in this file that the functions were available in 1.6 without doing the legwork to sort out what was going on, sorry about that :sheep: .